src/test/python/parser/opentrackio_lib.py

```diff
@@ -137,60 +234,66 @@ def get_timecode(self):
-            spd = 86400                         # sec per day
-            spy = 31536000                      # sec per year
-            # separate into years, days, hours, min, sec
-            ydelta = int(ssec / spy)            # years since epoch
-            yr = int(epoch + ydelta)            # current year
-            sty = int(ssec - (ydelta * spy))    # seconds elapsed this year
-            day = int(sty / spd)                # current day of year
-            std = int(sty - (day * spd))        # seconds elapsed today (since midnight)
-            hr = int(std / sph)                 # hours elapsed today
-            mn = int((std - (hr * sph)) / spm)  # remainder minutes
-            st = int(std - (hr * sph) - (mn * spm))  # remainder seconds
-        if not part:
-            if self.sample_time_format == "sec":
-                return ssec + (nsec * 0.000000001)
-            elif self.sample_time_format == "timecode":             # since midnight
-                frm = int((nsec * 0.000000001) * self.get_timecode_framerate())
-                return '{:02}'.format(hr) + ":" + '{:02}'.format(mn) + ":" + '{:02}'.format(st) + ":" + '{:02}'.format(frm)
-            elif self.sample_time_format == "string":
-                return "year:{} day:{} hour:{} min:{} sec:{} nsec:{}".format(yr,day,hr,mn,st,nsec)
-        else:
-            if part == 'yy':
-                return yr
-            elif part == 'dd':
-                return day
-            elif part == 'hh':
-                return hr
-            elif part == 'mm':
-                return mn
-            elif part == 'ss':
-                return st
-            elif part == 'ns':
-                return nsec
+        
+            # Calculate the total time as seconds plus fractional seconds from nanoseconds and attoseconds
+            total_time = ssec + (nsec * 1e-9)
+            base_time = datetime(1970, 1, 1) + timedelta(seconds=ssec, microseconds=nsec / 1_000)
+        
+            if not part:
+                t_format = self.sample_time_format
+                if format and format != self.sample_time_format:
+                    t_format = format
+                if t_format == TimeFormat.SECONDS:
+                    return total_time
+                elif t_format == TimeFormat.TIMECODE:
+                    frames = int((nsec * 1e-9) * self.get_timecode_framerate())
+                    return f"{base_time.hour:02}:{base_time.minute:02}:{base_time.second:02}:{frames:02}"
+                elif t_format == TimeFormat.STRING:
+                    return f"year:{base_time.year} day:{base_time.timetuple().tm_yday} hour:{base_time.hour} min:{base_time.minute} sec:{base_time.second} nsec:{nsec}"
+                elif t_format == TimeFormat.ISO8601:
+                    return base_time.strftime(f"%Y-%jT%H:%M:%S.{nsec}")
+            else:
+                # Return individual components based on 'part' argument
+                parts_map = {
```

Comment:
As above, this is now a Pythonic data struct, not as easy to port to C language. But if you think this is the way to go with examples, I can live with it.